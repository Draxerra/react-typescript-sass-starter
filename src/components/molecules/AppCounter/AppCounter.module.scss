.app-counter {
  display: grid;
  grid-template-columns: 1fr 1fr;
  place-items: center;
  gap: var(--spacing);
  &__text {
    font-size: var(--fs-500);
    grid-column: 1 / 3;
  }
  &__actions {
    display: flex;
    gap: var(--spacing);
  }
}
