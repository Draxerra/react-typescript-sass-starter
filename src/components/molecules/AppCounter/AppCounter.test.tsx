import AppCounter from "./AppCounter";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("App Counter", () => {
  it("should render", () => {
    const decrement = jest.fn();
    const increment = jest.fn();
    const counter = render(
      <AppCounter decrement={decrement} increment={increment}>
        10
      </AppCounter>
    ).asFragment();
    expect(counter).toMatchSnapshot();
  });

  it("should fire decrement on counter-decrement click", () => {
    const decrement = jest.fn();
    const increment = jest.fn();
    const { getByText } = render(
      <AppCounter decrement={decrement} increment={increment}>
        10
      </AppCounter>
    );
    const decrementBtn = getByText("Decrement");
    userEvent.click(decrementBtn);
    expect(decrement).toHaveBeenCalledTimes(1);
  });

  it("should fire icrement on counter-icrement click", () => {
    const decrement = jest.fn();
    const increment = jest.fn();
    const { getByText } = render(
      <AppCounter decrement={decrement} increment={increment}>
        10
      </AppCounter>
    );
    const incrementBtn = getByText("Increment");
    userEvent.click(incrementBtn);
    expect(increment).toHaveBeenCalledTimes(1);
  });
});
