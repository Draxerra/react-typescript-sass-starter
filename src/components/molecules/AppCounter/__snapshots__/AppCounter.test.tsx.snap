// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Counter should render 1`] = `
<DocumentFragment>
  <div
    class="app-counter"
  >
    <span
      class="app-counter__text"
    >
      Current Count: 10
    </span>
    <button
      aria-label="Increment value"
      class="counter-increment app-button"
    >
      Increment
    </button>
    <button
      aria-label="Decrement value"
      class="counter-decrement app-button"
    >
      Decrement
    </button>
  </div>
</DocumentFragment>
`;
